### PR TITLE
Locale improvements

### PR DESCRIPTION
**Pull request by marcoacierno**

### CODE REVIEW

File: frontend/src/locale/context.tsx [hunk 1]
```diff
@@ -22,8 +24,11 @@ export const LocaleProvider: React.FC<{ lang: string }> = ({
   children,
 }) => {
   React.useEffect(() => {
-    if (lang !== localStorage.getItem("locale")) {
-      localStorage.setItem("locale", lang);
+    const { pyconLocale } = cookies({
+      req: { headers: { cookie: document.cookie } },
+    });
+    if (lang !== pyconLocale) {
+      document.cookie = `pyconLocale=${lang}; path=/`;
```
Comment (patrick91) [inline frontend/src/locale/context.tsx:31]: why are you using cookies for the language?

Comment (marcoacierno) [reply frontend/src/locale/context.tsx:31]: we can access them on the server and redirect server side

Comment (marcoacierno) [reply frontend/src/locale/context.tsx:31]: otherwise we would have to load the JS, flash the UI for a second and then redirect to the correct path

Comment (patrick91) [reply frontend/src/locale/context.tsx:31]: ok, just saw it below :) we don't use the user language by default anymore?

Comment (marcoacierno) [reply frontend/src/locale/context.tsx:31]: I wanted to add it back in another PR so keep this one simple, I can do it in this one too

Comment (patrick91) [reply frontend/src/locale/context.tsx:31]: It’s ok to do in another PR :)

**Pull request merged**